### PR TITLE
initial checkin

### PR DESCRIPTION
**Pull request by SteveNY-Tibco**
Please review my operation modules

### CODE REVIEW

File: operations/cleaning/groupBy/operation.go [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+import (
+	"bytes"
+
+	"github.com/project-flogo/cml/action/operation"
+	"github.com/project-flogo/cml/operations/common"
+	"github.com/project-flogo/core/data/metadata"
+	"github.com/project-flogo/core/support/log"
+)
+
+func init() {
```
Comment (skothari-tibco) [inline operations/cleaning/groupBy/operation.go]: You'll need to remove the init() function from the operations

File: operations/cleaning/groupBy/go.mod [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+require (
+	github.com/project-flogo/cml/action v0.0.0-00010101000000-000000000000
+	github.com/project-flogo/cml/operations/common v0.0.0-00010101000000-000000000000
+	github.com/project-flogo/core v0.9.3
+	github.com/stretchr/testify v1.4.0
+)
+
+replace github.com/project-flogo/cml/action => ../../../action
+
+replace github.com/project-flogo/cml/operations/common => ../../common
```
Comment (skothari-tibco) [inline operations/cleaning/groupBy/go.mod:14]: You don't need this if you remove `go.mod` from common.

**Review by skothari-tibco — CHANGES_REQUESTED** [review]

File: operations/cleaning/groupBy/operation.go [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+	if err != nil {
+		return nil, err
+	}
+
+	var result interface{}
+
+	this.logger.Info("Input dataFrame is : ", in.Data)
+	this.logger.Info("Parameter is : ", this.params)
+
+	result, err = this.groupBy(in.Data.(map[string][]interface{}))
```
Comment (skothari-tibco) [inline operations/cleaning/groupBy/operation.go]: Can you change `this` to `opt` or `operation` ?

File: operations/cleaning/join/metadata.go [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+	var err error
+	i.Left, err = ToDataFrame(values["left"])
+	i.Right, err = ToDataFrame(values["right"])
+	i.LeftIndex, err = ToDataFrame(values["leftindex"])
+	i.RightIndex, err = ToDataFrame(values["rightindex"])
+
+	return err
+}
+
+func ToDataFrame(val interface{}) (interface{}, error) {
```
Comment (skothari-tibco) [inline operations/cleaning/join/metadata.go]: Isn't this unnecessary?

File: operations/cleaning/join/operation.go [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+	var result interface{}
+
+	this.logger.Info("Left dataFrame is : ", in.Left)
+	this.logger.Info("Right dataFrame is : ", in.Right)
+	this.logger.Info("Left Index map is : ", in.LeftIndex)
+	this.logger.Info("Right Index map is : ", in.RightIndex)
+	this.logger.Info("Parameter is : ", this.params)
+
+	result, err = this.join(
+		in.Left.(map[string][]interface{}),
```
Comment (skothari-tibco) [inline operations/cleaning/join/operation.go:44]: What if in.Left is of type `map[string]interface{}`, wouldn't this throw error ?

**Review by abramvandergeest — CHANGES_REQUESTED** [review]
So you have the function ToInterfaceArray in the common library, but you also define it again in the metadata.go file of both transpose and mean.  Could you just consolidate it down to the common library? Cheers

**Review by abramvandergeest — CHANGES_REQUESTED** [review]
Can you include a README of the style at "https://github.com/abramvandergeest/cml/blob/master/sample_README.MD" for each operation?  This is where you would include discussions like how axis behaves in mean.

**Comment by abramvandergeest** [comment]
Looks good to me.

**Review by abramvandergeest — APPROVED** [review]
Looks good to me

**Review by skothari-tibco — APPROVED** [review]

**Pull request merged**